controllers/datadogagent/feature/orchestrator/feature.go

```diff
@@ -0,0 +1,182 @@
+		reqComp.Agent = feature.RequiredComponent{
+			IsRequired: apiutils.NewBoolPointer(true),
+			Containers: []apicommonv1.AgentContainerName{apicommonv1.ProcessAgentContainerName},
+		}
+
+		if orchestrator.Conf != nil {
+			f.customConfig = v2alpha1.ConvertCustomConfig(orchestrator.Conf)
+		}
+		f.configConfigMapName = apicommonv1.GetConfName(dda, f.customConfig, apicommon.DefaultOrchestratorExplorerConf)
+		f.scrubContainers = apiutils.BoolValue(orchestrator.ScrubContainers)
+		f.extraTags = orchestrator.ExtraTags
+		if orchestrator.DDUrl != nil {
+			f.ddURL = *orchestrator.DDUrl
+		}
+		f.serviceAccountName = v2alpha1.GetClusterAgentServiceAccount(dda)
+
+		if v2alpha1.IsClusterChecksEnabled(dda) {
+			f.clusterChecksEnabled = true
+
+			if v2alpha1.IsCCREnabled(dda) {
+				f.rbacSuffix = common.ChecksRunnerSuffix
+				f.serviceAccountName = v2alpha1.GetClusterChecksRunnerServiceAccount(dda)
+				reqComp.ClusterChecksRunner.IsRequired = apiutils.NewBoolPointer(true)
+			}
+		}
+	}
+
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *orchestratorExplorerFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	f.owner = dda
+	orchestrator := dda.Spec.Features.OrchestratorExplorer
+
+	if orchestrator != nil && apiutils.BoolValue(orchestrator.Enabled) {
+		reqComp.ClusterAgent.IsRequired = apiutils.NewBoolPointer(true)
+		reqComp.Agent = feature.RequiredComponent{
+			IsRequired: apiutils.NewBoolPointer(true),
+			Containers: []apicommonv1.AgentContainerName{apicommonv1.ProcessAgentContainerName},
+		}
+
+		if orchestrator.Conf != nil {
+			f.customConfig = v1alpha1.ConvertCustomConfig(orchestrator.Conf)
+		}
+		f.configConfigMapName = apicommonv1.GetConfName(dda, f.customConfig, apicommon.DefaultOrchestratorExplorerConf)
+		if orchestrator.Scrubbing != nil {
+			f.scrubContainers = apiutils.BoolValue(orchestrator.Scrubbing.Containers)
+		}
+		f.extraTags = orchestrator.ExtraTags
+		if orchestrator.DDUrl != nil {
+			f.ddURL = *orchestrator.DDUrl
+		}
+		f.serviceAccountName = v1alpha1.GetClusterAgentServiceAccount(dda)
+
+		if v1alpha1.IsClusterChecksEnabled(dda) && apiutils.BoolValue(orchestrator.ClusterCheck) {
+			f.clusterChecksEnabled = true
+
+			if v1alpha1.IsCCREnabled(dda) {
+				reqComp.ClusterChecksRunner.IsRequired = apiutils.NewBoolPointer(true)
+
+				f.rbacSuffix = common.ChecksRunnerSuffix
+				f.serviceAccountName = v1alpha1.GetClusterChecksRunnerServiceAccount(dda)
+			}
+		}
+	}
+
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *orchestratorExplorerFeature) ManageDependencies(managers feature.ResourceManagers, components feature.RequiredComponents) error {
+	// Manage the Check Configuration in a configmap
+	configCM, err := f.buildOrchestratorExplorerConfigMap()
```

Comment:
i think we should rm these lines in the defaulting code so that the feature code can handle the default configmap build: https://github.com/DataDog/datadog-operator/blob/4467bbebd405d2d3a9f72f515ccc1ee2d27e37c7/apis/datadoghq/v2alpha1/datadogagent_default.go#L207-L211

(we just fixed a similar issue in the ksmcore feature)